### ISSUE TITLE
Rule Change: support TypeScript syntax in no-duplicate-imports

### ISSUE THREAD

**Issue by rivajunior** [eslint/eslint#19834]
### What rule do you want to change?

no-duplicate-imports

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/*eslint no-duplicate-imports: "error"*/
import { Foo } from './foo'
import type { Bar } from './foo' // error  './foo' import is duplicated  no-duplicate-imports
```

### What does the rule currently do for this code?

It currently reports an error in type imports separated in a different import statement

### What will the rule do after it's changed?

/*eslint no-duplicate-imports: ["error", { "allowTypeImports": true }]*/
import { Foo } from './foo'
import type { Bar } from './foo' // Ok!

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by Tanujkanti4441** [comment]
Hi @rivajunior, thanks for the suggestion.

```ts
/*eslint no-duplicate-imports: "error"*/
import { Foo } from './foo'
import type { Bar } from './foo'
```
Even though this can be written as:

```ts
/*eslint no-duplicate-imports: "error"*/
import { Foo, type Bar } from './foo'
```

I am fine with the option `allowTypeImports` here.

@eslint/eslint-team, should we consider this?

**Comment by nzakas** [comment]
I'm okay with adding an option. I don't think `allowTypeImports` accurately represents what's wanted here. It seems like we want to say, "It's okay to have duplicate imports so long as one imports values and one imports types." I don't think we want allow any number of `import type` declarations to the same package. So maybe `allowSeparateTypeImports`?

**Comment by rivajunior** [comment]
> I'm okay with adding an option. I don't think `allowTypeImports` accurately represents what's wanted here. It seems like we want to say, "It's okay to have duplicate imports so long as one imports values and one imports types." I don't think we want allow any number of `import type` declarations to the same package. So maybe `allowSeparateTypeImports`?

That's a good interpretation of the desired behavior. I agree with you that `allowSeparateTypeImports` is better. Thank you for the suggested option name.

**Comment by sethamus** [comment]
I'm in favor of `allowSeparateTypeImports`. I would also like to implement this if accepted.

**Comment by nzakas** [comment]
**TSC Summary:** The issue is requesting an enhancement to the no-duplicate-imports rule to better support TypeScript syntax. The current rule flags an error when TypeScript type imports are used alongside regular imports from the same module. Currently, the rule reports an error when a file includes both a value import and a type import from the same module.


**TSC Question:** Do we want to add the proposed option `allowSeparateTypeImports` that would allow separate import statements for types from the same module without triggering the "duplicate imports" rule error?

**Comment by Un5had0wed** [comment]
> I'm in favor of `allowSeparateTypeImports`. I would also like to implement this if accepted.

@sethamus By when are you gonna give this ? I have the fix ready. I would like to do this as my first contribution if you are ok with it

**Comment by sethamus** [comment]
@Un5had0wed It's done, but I need the issue to be accepted before I submit. As you can see, it has the TSC agenda label.

**Comment by sam3k** [comment]
We've resolved to accept the proposal for the `allowSeparateTypeImports` option to allow `import type` alongside `import` from the same module per TSC 2025-06-12 meeting.

**Comment by nzakas** [comment]
@sam3k thanks! When we've agreed to accept an issue, can you please add the "accepted" label? That will save us an extra step.

@sethamus all set!
